src/components/form/form.tsx

```diff
@@ -0,0 +1,124 @@
+import { Button, Field, FormGeneric, FormPanel } from '@mdb/flora';
+import {
+  jobFunctions,
+  institutionTypes,
+  teachingStatuses,
+} from 'src/lib/registration';
+import FormProps from './types';
+import styles from './styles';
+
+interface FieldInterface {
+  name: string;
+  label: string;
+  options?: Array<string>;
+  component: 'text-input' | 'select' | 'text-area' | 'checkbox';
+}
+
+const fields: Array<FieldInterface> = [
+  {
+    name: 'firstName',
+    label: 'First Name',
+    component: 'text-input',
+  },
+  {
+    name: 'lastName',
+    label: 'Last Name',
+    component: 'text-input',
+  },
+  {
+    name: 'email',
+    label: 'Email',
+    component: 'text-input',
+  },
+  {
+    name: 'jobFunction',
+    label: 'Your Job Function',
+    component: 'select',
+    options: jobFunctions,
+  },
+  {
+    name: 'teachingStatus',
+    label: 'Teaching Status',
+    component: 'select',
+    options: teachingStatuses,
+  },
+  {
+    name: 'institutionName',
+    label: 'Name of Institution',
+    component: 'text-input',
+  },
+  {
+    name: 'institutionType',
+    label: 'Institution Type',
+    component: 'select',
+    options: institutionTypes,
+  },
+  {
+    name: 'location',
+    label: 'Country',
+    component: 'select',
+    options: ['United States of America', 'Canada', 'Mexico'], // TODO: proper list of these - fetched data?
```

Comment:
Ah yeah so I was looking into this a while ago and https://github.com/esosedi/3166 seems like a solid source for these. I know the current devhub academia page is using https://www.npmjs.com/package/react-select-country-list, which is a little outdated but would give us parity?

But regardless I think we should pull from a list. May have to prop drill these down, or use https://www.npmjs.com/package/next-plugin-preval to populate the list at build time and just import it.